pyproject.toml

```diff
@@ -308,7 +308,7 @@ all = [
 [tool.poetry.dev-dependencies]
 # We pin black so that our tests don't start failing on new releases.
```

Comment:
This says that we pin, but we use `>=` below. I'm not quite sure what's up with that?

Reply:
That was probably me. I think the comment is from pre-poetry times when `setup.py` was the closest thing we had to a lockfile. It's now a little out of date; suggest removing the comment.

(One advantage of not pinning in pyproject.toml is that it'll make dependabot update the lockfile with a new `black` release if one comes out.)

Reply:
I came up with the same advantage, but want sure it was on purpose!

Reply:
#14227 apparently!

Reply:
Ahh, I think the point was that I wanted dependabot to be able to update e.g. `black` without editing pyproject.toml. (The lockfile includes a hash of pyproject.toml, and changes to that hash were frequently conflicting.)

Reply:
Maybe my comment isn't quite accurate then...